sql/size_utils.sql

```diff
@@ -682,3 +635,125 @@ BEGIN
 END;
 $BODY$;
 
+-------------Get index size for hypertables -------
+--schema_name      - schema_name for hypertable index
+-- index_name      - index on hyper table
+---note that the query matches against the hypertable's schema name as
+-- the input is on the hypertable index nd not the chunk index.
+CREATE OR REPLACE FUNCTION _timescaledb_internal.indexes_local_size(
+    schema_name_in             NAME,
+    index_name_in              NAME
+)
+RETURNS TABLE ( hypertable_id INTEGER,
+                total_bytes BIGINT ) 
+               LANGUAGE PLPGSQL STABLE STRICT
```

Comment:
i think this should be volatile instead of strict

Reply:
Could you elaborate why? this function does not mutate anything.

Reply:
because the output is not constant for the same input within the same session

Reply:
it only needs to be constant for the statement:  https://www.postgresql.org/docs/8.1/xfunc-volatility.html

Reply:
yes but that is not guaranteed either

Reply:
```
dev=# \df+ pg_relation_size
                                                                                                                                List of functions
   Schema   |       Name       | Result data type | Argument data types | Type | Volatility | Parallel |  Owner   | Security | Access privileges | Language |                  Source code                   |                            Description

------------+------------------+------------------+---------------------+------+------------+----------+----------+----------+-------------------+----------+------------------------------------------------+-----------------------------------------------------------------
---
 pg_catalog | pg_relation_size | bigint           | regclass            | func | volatile   | safe     | postgres | invoker  |                   | sql      | select pg_catalog.pg_relation_size($1, 'main') | disk space usage for the main fork of the specified table or ind
ex
 pg_catalog | pg_relation_size | bigint           | regclass, text      | func | volatile   | safe     | postgres | invoker  |                   | internal | pg_relation_size                               | disk space usage for the specified fork of a table or index
```

Reply:
Ah, I see that: select hypertable_index_size('idx'),  hypertable_index_size('idx) ? The 2 function calls could return different values.

Reply:
^ Looks like the messages crossed and we agree. Modified to volatile.